privacyidea/lib/resolvers/HTTPResolver.py

```diff
@@ -0,0 +1,229 @@
+
+    @staticmethod
+    def getResolverDescriptor():
+        """
+        return the descriptor of the resolver, which is
+        - the class name and
+        - the config description
+
+        :return: resolver description dict
+        :rtype:  dict
+        """
+        return HTTPResolver.getResolverClassDescriptor()
+
+    def getUserId(self, loginName):
+        """
+        This method only echo's the loginName parameter
+        """
+        return loginName
+
+    def getUsername(self, userid):
+        """
+        This method only echo's the userid parameter
+        """
+        return userid
+
+    def getUserInfo(self, userid):
+        """
+        This function returns all user information for a given user object
+        identified by UserID.
+        :param userid: ID of the user in the resolver
+        :type userid: int or string
+        :return:  dictionary, if no object is found, the dictionary is empty
+        :rtype: dict
+        """
+        return HTTPResolver._getUser(self.config, userid)
+
+    def getUserList(self, searchDict=None):
+        """
+        Since it is an HTTP resolver,
+        users are not stored in the database
+        """
+        return []
+
+    def getResolverId(self):
+        """
+        get resolver specific information
+        :return: the resolver identifier string - empty string if not exist
+        """
+        return self.config['endpoint'] if 'endpoint' in self.config else ''
+
+    def loadConfig(self, config):
+        """
+        Load the configuration from the dict into the Resolver object.
+        If attributes are missing, need to set default values.
+        If required attributes are missing, this should raise an
+        Exception.
+
+        :param config: The configuration values of the resolver
+        :type config: dict
+        """
+        self.config = config
+        return self
+
+    @classmethod
+    def testconnection(cls, param):
+        """
+        This function lets you test if the parameters can be used to create a
+        working resolver.
+        The implementation should try to connect to the user store and verify
+        if users can be retrieved.
+        In case of success it should return a text like
+        "Resolver config seems OK. 123 Users found."
+
+        :param param: The parameters that should be saved as the resolver
+        :type param: dict
+        :return: returns True in case of success and a descriptive text
+        :rtype: tuple
+        """
+        desc = ""
+        success = False
+        try:
+            response = cls._getUser(param, param.get('testEmail'))
+            desc = response
+            success = True
+        except Exception as e:
+            success = False
+            desc = "failed: {0!s}".format(e)
+        return success, desc
+
+    #
+    #   Private methods
+    #
+    @classmethod
+    def _getUser(cls, param, userid):
+        method = param.get('method').lower()
```

Comment:
I didn't find a clean way to do this based on other resolvers. But I can add some validations, no problem.

Reply:
we have the [`getParam()`](https://github.com/privacyidea/privacyidea/blob/5fe428af3c75cd1c0013b244950fd2abd98b950d/privacyidea/api/lib/utils.py#L55) function which can throw an error if a required parameter isn't given.

Reply:
Ok, I'm going to use it. Also, let me show you something:

![image](https://user-images.githubusercontent.com/2397875/88084585-04ba4480-cb5b-11ea-89e9-c3f876fb49d7.png)

If I set `allowed_values=(get, post)` and optional as False, `getParam()` should throw an error if I send for example "put". Now, this method puts the default value instead of throwing an exception. Is it expected?